src/extension.ts

```diff
@@ -108,6 +108,23 @@ class RunFormatOnSave {
 		return process.execFile(executable, args, {cwd: this.projectDir});
 	}
 
+	getProjectPath(contentRoot : string, fileName : string) : string {
```

Comment:
I did this pretty quickly to just get a POC out - in reality we'd probably want to use Node's [path parse API](https://nodejs.org/api/path.html#path_path_parse_path).